tests/clients/test_alerts.py

```diff
@@ -284,21 +284,52 @@ def test_get_all_posts_expected_data(self, mock_session, user_context):
             and posted_data["srtDirection"] == "ASC"
         )
 
-    def test_get_by_name_filters_correct_record(self, mock_get_all_session, user_context):
-        alert_client = AlertClient(mock_get_all_session, user_context)
-        rule = alert_client.get_rules_by_name(u"TESTNAME")
-        assert len(rule) == 2
-
-    def test_get_by_name_filters_correct_record_case_insenstive_search(
-        self, mock_get_all_session, user_context
+    def test_get_all_rules_by_name_posts_expected_data(
+        self, mock_session, user_context, successful_post
     ):
-        alert_client = AlertClient(mock_get_all_session, user_context)
-        rule = alert_client.get_rules_by_name(u"TestName")
-        assert len(rule) == 2
+        alert_client = AlertClient(mock_session, user_context)
+        for _ in alert_client.get_all_rules_by_name(
+            "testname", sort_key="key", sort_direction="ASC"
+        ):
+            break
 
-    def test_get_by_name_raises_exception_when_name_does_not_match(
-        self, mock_get_all_session, user_context
+        assert mock_session.post.call_count == 1
+        posted_data = json.loads(mock_session.post.call_args[1]["data"])
+        filter_group = posted_data["groups"][0]["filters"][0]
+
+        assert filter_group["term"] == "Name"
+        assert filter_group["operator"] == "IS"
+        assert filter_group["value"] == "testname"
+        assert mock_session.post.call_args[0][0] == "/svc/api/v1/rules/query-rule-metadata"
+        assert (
+            posted_data["tenantId"] == user_context.get_current_tenant_id()
+            and posted_data["groupClause"] == "AND"
+            and posted_data["pgNum"] == 0
+            and posted_data["pgSize"] == 1000
+            and posted_data["srtKey"] == "key"
+            and posted_data["srtDirection"] == "ASC"
+        )
+
+    def test_get_rule_by_observer_id_posts_expected_data(
+        self, mock_session, user_context, successful_post
     ):
-        alert_client = AlertClient(mock_get_all_session, user_context)
-        rule = alert_client.get_rules_by_name(u"TESTNAME2")
-        assert len(rule) == 0
+        alert_client = AlertClient(mock_session, user_context)
+        for _ in alert_client.get_rule_by_observer_id("testid"):
+            break
+
+        assert mock_session.post.call_count == 1
+        posted_data = json.loads(mock_session.post.call_args[1]["data"])
+        filter_group = posted_data["groups"][0]["filters"][0]
+
+        assert filter_group["term"] == "ObserverRuleId"
+        assert filter_group["operator"] == "IS"
+        assert filter_group["value"] == "testid"
+        assert mock_session.post.call_args[0][0] == "/svc/api/v1/rules/query-rule-metadata"
+        assert (
+            posted_data["tenantId"] == user_context.get_current_tenant_id()
+            and posted_data["groupClause"] == "AND"
+            and posted_data["pgNum"] == 0
```

Comment:
Have we tested these to make sure that pages start at 0 and not 1 ?

Reply:
it uses the same endpoint as stuff where we know that to be the case, and i've also physically tested this, so yes